### DISCUSSION TITLE
Existe alguma biblioteca que reproduza este filtro de tabela  no React  ?

### DISCUSSION THREAD

**Discussion by ViniciusGularte** [frontendbr/forum#1489] [Perguntas e Respostas]
Bom dia galera, estou com uma pedra grande no meu desenvolvimento, preciso do filtro abaixo na minha aplicação, no caso criaria um array único com base em cada atributo dos objetos recebidos , e  usaria como filtro como na imagem abaixo:
![2019-07-15_11-15](https://user-images.githubusercontent.com/25805846/61223006-36036780-a6f2-11e9-8ffa-3925ef0331ba.png)

No caso esta imagem é de uma lib em jquery que trata o html, eu recriei a  mesma para tratar os objetos recebidos da API, porém a logica ficou grande em demasia(umas 300 linhas só de logica pra tratar os dados e enviar para um outro componente que gera os filtros e tabela, por componente(Multiple isto por uns 30 componentes  :dizzy_face:), alguém já teve que criar algo parecido em um Spa? Queria deixar pra ultimo caso usar jQuery no projeto e ficar tratando html.

**Comment by fdaciuk** [comment]
Opa! Nesse vídeo eu mostro um filtro, mas usando um input de texto.. a ideia é parecida com o que você quer fazer:
https://youtu.be/my9wmfOkIP4

**Comment by ViniciusGularte** [comment]
Opa @fdaciuk  tudo bem?  este tipo de filtro consegui criar de boa :D filtrando todos atributos do objeto a partir da string recebida do input de texto, eu queria saber se alguém já teve que enfrentar este tipo de filtro no React , onde eu tenho um estado de filtros, e a partir de cada check box ele filtra o objeto inteiro, qualquer coisa acho que vou ter que criar minha própria lib pra isto kkk

**Comment by ViniciusGularte** [comment]
Consegui, no caso criei um fork da lib [React Table Select](https://github.com/cheekujha/react-table-filter), e estou conseguindo modificar para o meu uso,  deixo o nome pra quem precisar deste tipo de filtro para sua aplicação. Vlwws

**Comment by fdaciuk** [comment]
Ah, saquei! Legal que conseguiu resolver :)
Pode deixar a issue aberta, assim se mais alguém tiver essa dúvida, pode encontrar a resposta aqui :)

**Comment by JudahAragao** [comment]
Acho que dessa forma eu já teria a possibilidade de criar filtros para gráficos com a lib recharts e outros. No caso, além de filtros únicos em cada gráfico, penso que dê para criar um filtro global com categorias para definir ano, setor, demanda. 

Funcionaria assim: se eu definir apenas uma data ele mostra a estatística de todos os setores e demandas conforme o ano definido, se apenas setor for definindo será mostrado a estatística de todas as demandas de todos os anos registrados sobre aquele determinado setor, se demanda e setor for definido será mostrado as estatísticas daquela demanda específica e daquele setor específico em todos os anos registrados não definidos e assim sucessivamente.

**Reply by JudahAragao** [reply]
Só agora que ví essa issue é de 2019. 
Sorry!

**Reply by censuradho** [reply]
eu ja ia comentar também kkk

**Reply by ViniciusGularte** [reply]
old but gold  😅 aquele react table select funcionou bem, sei que ta até hoje rodando  no sistema dos caras o fork que tinha criado
